added User entity and related classes & interfaces:

DanilKi opened the pull request:
- added DTOs for handling http requests & responses;
 - added repository, mapper, service for User entity;
 - added AuthController with registration endpoint;
 - added RegistrationException and its handling method to GlobalExceptionHandler;
 - added FieldMatch annotation for validating matches of two fields of an object.

DanilKi committed:
added User entity and related classes & interfaces:
 - added DTOs for handling http requests & responses;
 - added repository, mapper, service for User entity;
 - added AuthController with registration endpoint;
 - added RegistrationException and its handling method to GlobalExceptionHandler;
 - added FieldMatch annotation for validating matches of two fields of an object.

Elena-Bruyako commented on src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -53,6 +53,18 @@ protected ResponseEntity<Object> handleEntityNotFound(RuntimeException ex,
+        body.put(HTTP_STATUS, HttpStatus.NOT_ACCEPTABLE);
+        body.put(ERRORS_MSG, ex.getLocalizedMessage());
+        return handleExceptionInternal(ex, body, new HttpHeaders(),
+                HttpStatus.NOT_ACCEPTABLE, request);
```
```suggestion
                HttpStatus.CONFLICT, request);
```

Elena-Bruyako commented on src/main/java/com/onlinebookstore/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
It's better to use `boolean existsByEmail(String email)`

Elena-Bruyako commented on src/main/resources/db/changelog/changes/02-create-users-table.yml:
```diff
@@ -0,0 +1,39 @@
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: email
```
Be sure, that the size in the same as in DTO fields

Elena-Bruyako reviewed (changes requested):
Good job! Let's fix some mistakes

DanilKi committed:
fixed following mistakes:
 - changed http-response status code to 409 'Conflict' when handling registration exception;
 - replaced findByEmail() with existsByEmail() method in user repository;
 - corrected corresponding sizes of fields in DB & DTO for User entity.
